SilverSolver/assignment4/islands.py

```diff
@@ -0,0 +1,44 @@
+from collections import deque
+from typing import List
+
+def neighbourhood(i: int, j: int, islandMatrix: List[List[int]]) -> List[List[int]]:
+    # helper function, returns the neighbourhood of (i, j) cell in islandMatrix
+    neigh = []
```

Comment:
for di, dj in ((-1, 0), (1, 0), (0, -1), (0, 1)):  # di, dj for deltai deltaj
  ni, nj = di + i, dj + j  # ni, nj is a neighbor candidate for i, j
  if 0 <= ni < len(islandMatrix) and 0 <= nj < len(islandMatrix[0]) and islandMatrix[ni][nj]:
    yield ni, nj

I find this version of neighbor a bit easier to read: 
- the neighbor candidate are more easily identified at the top of the for loop,
- the guards that decide whether a candidate is good or not are clearly shown after an 'if' clauses
- we avoid the repetition of the same pattern of code which could have a slight typos
- nit: no need to create and empty list, yield saves two lines, and is slightly more efficient